infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -393,10 +365,25 @@ private void applyShadowStream(UUID streamId, long snapshot) {
      * Read from shadowStream and append/apply to the actual stream
      */
     public void applyShadowStreams() {
-        log.debug("Apply Shadow Streams, total={}", streamViewMap.size());
-
+        log.debug("Apply Shadow Streams, total={}", config.getStreamMap().size());
         long snapshot = rt.getAddressSpaceView().getLogTail();
-        for (UUID regularStreamId : streamViewMap.keySet()) {
+
+        // Apply registry table first for streaming consideration in DYNAMIC mode.
+        if (config.getStreamMap().containsKey(REGISTRY_TABLE_ID)) {
```

Comment:
In the apply phase, we do not require any validation with the streams present in the registry table.  So we do not need to apply the table first.  The sync can be performed after apply finishes.

Reply:
For this clause, applying the registry table is for streaming purposes. i.e. If Source has opened some tagged tables that Sink hasn't opened... So I think we need to apply registry table first and sync the config, then applying other tables

Reply:
got it, yes.  Please add a comment for this